### PR TITLE
Add OpenTracing support

### PR DESCRIPTION
**Pull request by alesj**
Introducing OpenTelemetry tracing option.

The new tracing module abstracts away the tracing usage -- either previous OpenTracing or new OpenTelemetry support. Instead of same tracing config code being applied over all test-client modules, we handle this in a single tracing module.
Different tracing implementations are triggered by setting implementation specific service name environment variable (see tracing's README for more details).
Each tracing implementation sets a global Tracer instance, which can then be used in supported libraries to provide new traces / spans.
Additionally to service name env var, other required env vars must be set as well -- see README.
Currently tracing implementations add support for Kafka consumer/producer and Kafka Streams tracing.

Similar implementation is used in Strimzi Operator and Strimzi Bridge,
so it might be useful to reconsider a better place for all this implementation code / artifact.

Signed-off-by: Ales Justin ales.justin@gmail.com

### CODE REVIEW

**Comment by alesj** [comment]
@im-konge can you please check / review this, but not yet merge -- I need to run the full test cycle ...

**Review by im-konge — APPROVED** [review]
Fine by me, assuming it works.

**Comment by alesj** [comment]
OK, full test cycle passing ... ready to merge.

**Comment by im-konge** [comment]
Let's wait for @scholzj or @Frawless for review

**Comment by scholzj** [comment]
@im-konge I think my comments from the previous PR still stand.

**Comment by im-konge** [comment]
@alesj can you please fix the comments which @scholzj  had on previous PR?

File: kafka/producer/src/main/java/ProducerConfiguration.java
Comment (im-konge) [inline kafka/producer/src/main/java/ProducerConfiguration.java:102]: Can you return it to `(config.getOauthAccessToken() != null)` please?

File: kafka/streams/src/main/java/StreamsConfiguration.java
Comment (im-konge) [inline kafka/streams/src/main/java/StreamsConfiguration.java:100]: Can you return it to previous state? (with parentheses)

**Review by im-konge — COMMENTED** [review]
Just two nits, that are connected to Jakub's comments

**Review by Frawless — APPROVED** [review]
In case it's working I am fine with that. And when you work-in comments from Jakub.

**Comment by alesj** [comment]
I resolved most of the questions / issues in the old PR, apart from these:

```
docker-images/build-images.sh
Outdated
echo "Installing root pom"
make mvn_build

echo "Building Tracing dependency"
make build --directory=tracing/
```

```
@scholzj on 23 Nov 2021
Normally, the submodule would be added into the SUBDIRS list in the main Makefile. So when you run make ... it would trigger it for the tracing module as well and build it that way.

The README.md suggests to call make all to build the project. But if this is only in the build-images.sh script, it would be never called and I assume it would not work with it. That is at least my read on the build.

 @alesj alesj on 23 Nov 2021
Atm the script iterates over kafka and http subdirs, and builds them.
Whereas tracing module is a seperate module on which they depend on,
so it must be built before.

This build-images.sh is only used in CI, afais.
The plain Make makes use of SUBDIRS, where tracing is listed as well.

If @im-konge or @Frawless have any better ideas, fine by me.
```

and

```
http/pom.xml
                <version>${opentracing-kafka.version}</version>
                <groupId>io.strimzi</groupId>
                <artifactId>test-client-tracing</artifactId>
                <version>${project.version}</version>
```

@scholzj Well, It didn't had test-client-tracing dependency from the same project before. So I think you can remove it here.
@alesj So it should go into each submodule?

File: pom.xml
Comment (im-konge) [inline pom.xml:31]: Why do you have two `jackson` versions?
```
         <jackson.version>2.12.4</jackson.version>
         ...
          <jackson-core.version>2.12.4</jackson-core.version>
```

Comment (alesj) [reply pom.xml]: Ah, good catch, removed.

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:40]: Cant it just use:
```
log.info("Initializing Jaeger ({}) tracingConfig with service name {}",
                        instance.type(),
                        serviceName)
```
?

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:18]: We would normally use uppercase for final static constants. You seem to use it for `HANDLES` as well.

@im-konge This is ideally something what checkstyle should check.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTracingHandle.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTracingHandle.java:30]: Define the string as `static final` constant.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:50]: Define the string as `static final` constant.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:43]: Define the string as `static final` constant.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:55]: Define the string as `static final` constant.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:38]: Define the string as `static final` constant.

File: kafka/producer/src/main/java/Main.java
Comment (scholzj) [inline kafka/producer/src/main/java/Main.java:25]: Why did the exception it throws changed?

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java
Comment (im-konge) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:18]: Sure, will add it there. Good catch

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:40]: Fixed.

**Review by scholzj — APPROVED** [review]
LGTM assuming it works.

**Pull request merged**
